libs/shell/shell.c

```diff
@@ -0,0 +1,209 @@
+#include <stdio.h>
+#include <string.h>
+
+#include <FreeRTOS.h>
+#include <task.h>
+#include <queue.h>
+#include <semphr.h>
+
+#include <stdbool.h>
+#include <util.h>
+#include <stdio.h>
+#include <shell.h>
+
+char *shell_prompt_text = "> ";
+char *shell_banner = "\n\nCommand Shell";
+
+struct shell_state {
+	/* +1 for the help command */
+	struct shell_cmd commands[SHELL_MAX_COMMANDS + 1];
+	int command_count;
+	char input_buf[32];
+	int input_pos;
+	bool initialized;
+	xSemaphoreHandle shell_lock;
+} shell_state;
+
+static bool shell_is_whitespace(char c)
+{
+	switch(c) {
+	case ' ':
+	case '\t':
+	case '\r':
+	case '\n':
+		return true;
+	default:
+		break;
+	}
+	return false;
+}
+
+static void shell_exec_cmd(int argc, char **argv)
+{
+	unsigned int cmd_len;
+	unsigned int tgt_cmd_len = strlen(argv[0]);
+	int i;
+	for (i = 0; i < shell_state.command_count; i++) {
+		cmd_len = strlen(shell_state.commands[i].command);
```

Comment:
The body of the loop might look cleaner if you assign shell_state.commands[i] to a const struct shell_cmd * local variable.